Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -581,7 +588,7 @@ private class MutableDictionary: BSONValue {
 
     /// methods required by the BSONValue protocol that we don't actually need/use. MutableDictionary
     /// is just a BSONValue to simplify usage alongside true BSONValues within the encoder.
-    required init(from iter: DocumentIterator) {
+    public static func from(iterator iter: DocumentIterator) -> BSONValue {
         fatalError("`MutableDictionary` is not meant to be initialized from a `DocumentIterator`")
```

Comment:
Any reason we use `fatalError()` here instead of preconditionFailure? Grepping through the driver, it looks like this is only used for MutableArray, Mutable Dictionary, and _BSONReferencingEncoder's deinitializer. It's outside the scope of this PR, but just something I'm curious about now that I've noticed it. Probably more a question for @kmahar.

Reply:
The only difference between fatalError vs preconditionFailure that I know of is that one of them is silent in release builds?

My guess is that @kmahar followed the Encoder & Decoder implementations from stdlib and they use fatalError here.

Reply:
Makes sense to me.